src/webapp/components/search-input/SearchInput.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { TextField } from "@material-ui/core";
+import { IconSearch24 } from "@dhis2/ui";
+import styled from "styled-components";
+import _ from "lodash";
+
+import i18n from "../../../utils/i18n";
+
+interface SearchInputProps {
```

Comment:
I hadn't realized that I've gotten into the habit of using interface instead of type in props, but I've seen that we tend to use interface, although in some places I have seen type. Maybe this is something we should talk about as a team to choose one and being more consistent

Reply:
I remember @tokland talked about this in a previous coding dojo, maybe it will be useful to write it in [know-how](https://github.com/EyeSeeTea/know-how/wiki)

Reply:
totally agree